core/keyboard_nav/ast_node.js

```diff
@@ -444,125 +448,128 @@ Blockly.ASTNode.prototype.navigateBetweenStacks_ = function(forward) {
- * @param {!Blockly.Block} block The source block of the current location.
- * @return {Blockly.ASTNode} An AST node pointing to the first field or input.
+ * @param {!Block} block The source block of the current location.
+ * @return {ASTNode} An AST node pointing to the first field or input.
  * Null if there are no editable fields or inputs with connections on the block.
  * @private
  */
-Blockly.ASTNode.prototype.findFirstFieldOrInput_ = function(block) {
-  var inputs = block.inputList;
-  for (var i = 0, input; (input = inputs[i]); i++) {
-    var fieldRow = input.fieldRow;
-    for (var j = 0, field; (field = fieldRow[j]); j++) {
-      if (field.isClickable() || Blockly.ASTNode.NAVIGATE_ALL_FIELDS) {
-        return Blockly.ASTNode.createFieldNode(field);
+ASTNode.prototype.findFirstFieldOrInput_ = function(block) {
+  const inputs = block.inputList;
+  let i = 0, input;
+  for (; (input = inputs[i]); i++) {
+    const fieldRow = input.fieldRow;
+    let j = 0, field;
+    for (; (field = fieldRow[j]); j++) {
+      if (field.isClickable() || ASTNode.NAVIGATE_ALL_FIELDS) {
+        return ASTNode.createFieldNode(field);
       }
     }
     if (input.connection) {
-      return Blockly.ASTNode.createInputNode(input);
+      return ASTNode.createInputNode(input);
     }
   }
   return null;
 };
 
 /**
  * Finds the source block of the location of this node.
- * @return {Blockly.Block} The source block of the location, or null if the node
+ * @return {Block} The source block of the location, or null if the node
  * is of type workspace.
  */
-Blockly.ASTNode.prototype.getSourceBlock = function() {
-  if (this.getType() === Blockly.ASTNode.types.BLOCK) {
-    return /** @type {Blockly.Block} */ (this.getLocation());
-  } else if (this.getType() === Blockly.ASTNode.types.STACK) {
-    return /** @type {Blockly.Block} */ (this.getLocation());
-  } else if (this.getType() === Blockly.ASTNode.types.WORKSPACE) {
+ASTNode.prototype.getSourceBlock = function() {
+  if (this.getType() === ASTNode.types.BLOCK) {
+    return /** @type {Block} */ (this.getLocation());
+  } else if (this.getType() === ASTNode.types.STACK) {
+    return /** @type {Block} */ (this.getLocation());
+  } else if (this.getType() === ASTNode.types.WORKSPACE) {
     return null;
   } else {
-    return /** @type {Blockly.IASTNodeLocationWithBlock} */ (
-      this.getLocation()).getSourceBlock();
+    return /** @type {IASTNodeLocationWithBlock} */ (this.getLocation())
+        .getSourceBlock();
   }
 };
 
 /**
  * Find the element to the right of the current element in the AST.
- * @return {Blockly.ASTNode} An AST node that wraps the next field, connection,
+ * @return {ASTNode} An AST node that wraps the next field, connection,
  *     block, or workspace. Or null if there is no node to the right.
  */
-Blockly.ASTNode.prototype.next = function() {
+ASTNode.prototype.next = function() {
+  let connection, block, nextConnection, targetConnection;
```

Comment:
I think declaring multiple variables in one line is disallowed: https://google.github.io/styleguide/jsguide.html#features-one-variable-per-declaration (which might conflict with the array advice..... in which case I would probably convert those arrays to more normal style of arrays `for (let i = 0, i < a.length; i++) {`)

Reply:
Noted. I try something else that follows the style guide. I was getting a different warning about declaring variables in a case block before, which is why i put them all up here.

Reply:
If it helps, you can (and [are allowed](https://google.github.io/styleguide/jsguide.html#formatting-switch-statements)) to put a block in each `case`:
```JS
switch (x) {
  case y: {
    // body;
    break;  // or return;
  }
  // etc.
}
```
…though admittedly it does look a little weird.

Reply:
I've added blocks to each `case` that require lexical scoping, but I'm not sure if I should add it to all cases if some of them need it.

For now I've only added it to those that need it, but I could add to the rest.